app/views/products/show.html.erb

```diff
@@ -72,14 +72,16 @@
     </div>
     <div class="product page-flex">
       <% @products.each do |product| %>
-        <div class="product-item">
-          <% product.photos.each do |photo| %>
-            <%= link_to image_tag(photo), product, :class => 'product-img' %>
-          <% end %>
-          <h3 class="product-name"><%= product.name %></h3>
-          <div class="product-price">$<%= product.price %></div>
-          <button class="product-add">ADD TO CART</button>
-        </div>
+        <% if @product.category == product.category && @product.name != product.name %>
```

Comment:
Instead of putting logic on view, you should move it to controller

For example
in controller: @related_products = Product.where(....)

in view: @related_products.each do .....